src/javascript/app_2/pages/portfolio/portfolio.jsx

```diff
@@ -0,0 +1,315 @@
+import React                from 'react';
+import moment               from 'moment';
+import PortfolioCard        from './portfolio_card.jsx';
+import DAO                  from '../../data/dao';
+import DataTable            from '../../components/elements/data_table.jsx';
+import Tooltip              from '../../components/elements/tooltip.jsx';
+import { getAppId }         from '../../../config';
+import ClientBase           from '../../../_common/base/client_base';
+import { formatMoney }      from '../../../_common/base/currency_base';
+import { localize }         from '../../../_common/localize';
+import { getPropertyValue } from '../../../_common/utility';
+import Loading              from '../../../../templates/_common/components/loading.jsx';
+
+const formatPortfolioData = (portfolio_arr) => {
+    const formatted_portfolio = portfolio_arr.map((portfolio_item) => {
+        const date_obj        = new Date(portfolio_item.expiry_time* 1000);
+        const moment_obj      = moment.utc(date_obj);
+        const remaining_time  = `${moment_obj.fromNow(true)} - ${moment_obj.format('h:mm:ss')}`;
+        const purchase        = parseFloat(portfolio_item.buy_price);
+        const payout          = parseFloat(portfolio_item.payout);
+        return {
+            reference: {
+                transaction_id: portfolio_item.transaction_id,
+                app_id        : portfolio_item.app_id,
+            },
+            type      : portfolio_item.contract_type,
+            details   : localize(portfolio_item.longcode.replace(/\n/g, '<br />')),
+            purchase  : formatMoney(false, purchase, true),
+            payout    : formatMoney(false, payout, true),
+            remaining_time,
+            id        : portfolio_item.contract_id,
+            indicative: {
+                amount: '',
+                style : '',
+            },
+        };
+    });
+    return formatted_portfolio;
+};
+
+// TODO: move to common
+const contract_type_display = {
+    ASIANU      : localize('Asian Up'),
+    ASIAND      : localize('Asian Down'),
+    CALL        : localize('Higher'),
+    CALLE       : localize('Higher or equal'),
+    PUT         : localize('Lower'),
+    DIGITMATCH  : localize('Digit Matches'),
+    DIGITDIFF   : localize('Digit Differs'),
+    DIGITODD    : localize('Digit Odd'),
+    DIGITEVEN   : localize('Digit Even'),
+    DIGITOVER   : localize('Digit Over'),
+    DIGITUNDER  : localize('Digit Under'),
+    EXPIRYMISS  : localize('Ends Outside'),
+    EXPIRYRANGE : localize('Ends Between'),
+    EXPIRYRANGEE: localize('Ends Between'),
+    LBFLOATCALL : localize('Close-Low'),
+    LBFLOATPUT  : localize('High-Close'),
+    LBHIGHLOW   : localize('High-Low'),
+    RANGE       : localize('Stays Between'),
+    UPORDOWN    : localize('Goes Outside'),
+    ONETOUCH    : localize('Touches'),
+    NOTOUCH     : localize('Does Not Touch'),
```

Comment:
is it best to localize each separately here? or localize them at the end after fetching them? up to you, I'm not sure either

Reply:
but I'm afraid localizing them like this won't work, it might get called before localize is populated

Reply:
Oh yeah that is correct, localizing them there won't work. Will update it 👍